### PR TITLE
feat: medic rework

### PR DESCRIPTION
**Pull request by Golbolco**
right now, Medic is a doctor that can self-visit

Medic now has a function comparable to the original idea for EM's Surgeon: it visits two players. if the first player dies, the second player gains an extra life

### CODE REVIEW

File: Games/types/Mafia/roles/cards/CanVisitSelf.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const Role = require("../../Card");
+
+module.exports = class CanVisitSelf extends Card {
+    constructor(role) {
+        super(role);
+
+    this.meetingMods = {
+      "*": {
+        targets: { include: ["alive"], exclude: [] },
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/CanVisitSelf.js:9]: canvisitself is not an appropriate name, 

e.g. roles like hooker currently visit non-mafia only

canvisitself + hooker implies it's non-mafia + self, when here it's everyone

suggestion: "can visit everyone"

Comment (Golbolco) [reply Games/types/Mafia/roles/cards/CanVisitSelf.js:9]: nvm, the whole card/modifier is superfluous anyway. will remove; rest of PR should be ready

File: Games/types/Mafia/roles/cards/DonateLife.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    this.meetings = {
+      "Harvest Life": {
+        states: ["Night"],
+        flags: ["voting"],
+        targets: { include: ["alive"], exclude: ["dead", "self"] },
+        action: {
+          priority: PRIORITY_NIGHT_SAVER - 1,
+          run: function () {
+            if (!this.player.alive) {
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/DonateLife.js:16]: wont work, people only die at priority 0

Comment (pikulet) [reply Games/types/Mafia/roles/cards/DonateLife.js:16]: the simplest way to fix this is to give the extra life after the death
i.e. extra life does not work tonight

Comment (Golbolco) [reply Games/types/Mafia/roles/cards/DonateLife.js:16]: fixed, also changed second action to an effect_giver priority

**Comment by pikulet** [comment]
@Golbolco you don't have to always merge master

**Review by pikulet — APPROVED** [review]

**Pull request merged**